### PR TITLE
feat(main)!: Astrocommunity compatible with v4

### LINKED ISSUE #463: lsp.inc-rename-nvim not working
`lsp.inc-rename-nvim` is not working with astrocommunity. `vim.lsp.buf.rename()` is called instead.
My workaround is setting the mapping to `<leader>rn`, since using `<leader>lr` does not work.
There is the following lsp_mapping in `.nvim/lua/astronvim/utils/lsp.lua`
Screenshot 2023-07-19 at 20 20 19
`inc-rename-nvim` does not override that mapping.

I see two possible solutions:
1. map `inc-rename-nvim` to `<leader>rn`
2. disable `vim.lsp.buf.rename()` mapping in the `init.lua` of `inc-rename-nvim`
 
Let me know which solution is better and I will raise PR 🙏

### PR DESCRIPTION
**Pull request by Uzaaft**
# Changelog
### Fixes
- Fix inc-rename. Closes #463

### Features
- rust: Remove rust-tools and use ferris.nvim instead. 
- json: Add schemastore to json pack
- yaml: Add schemastore to yaml pack

### CODE REVIEW

File: lua/astrocommunity/bars-and-lines/heirline-mode-text-statusline/init.lua [hunk 1]
```diff
@@ -1,7 +1,7 @@
 return {
   "rebelot/heirline.nvim",
   opts = function(_, opts)
-    local status = require "astronvim.utils.status"
+    local status = require "astrocore.status"
```
Comment (kshlm) [inline lua/astrocommunity/bars-and-lines/heirline-mode-text-statusline/init.lua:4]: Astroui has status now.
```suggestion
    local status = require "astroui.status"
```

File: lua/astrocommunity/bars-and-lines/heirline-vscode-winbar/init.lua [hunk 2]
```diff
@@ -1,7 +1,7 @@
 return {
   "rebelot/heirline.nvim",
   opts = function(_, opts)
-    local status = require "astronvim.utils.status"
+    local status = require "astrocore.status"
```
Comment (kshlm) [inline lua/astrocommunity/bars-and-lines/heirline-vscode-winbar/init.lua:4]:
```suggestion
    local status = require "astroui.status"
```

File: lua/astrocommunity/pack/rust/init.lua [hunk 3]
```diff
@@ -1,21 +1,63 @@
...
+    end,
+  },
+  {
+    "mrcjkb/rustaceanvim",
+    version = "^3",
+    ft = "rust",
     opts = function()
       local adapter
       local success, package = pcall(function() return require("mason-registry").get_package "codelldb" end)
+      local cfg = require "rustaceanvim.config"
```
Comment (mrcjkb) [inline lua/astrocommunity/pack/rust/init.lua:60]: rustaceanvim auto-detects `codelldb` from the path and from the mason-registry (if installed).
So this should now be redundant :smile:

Comment (Uzaaft) [reply lua/astrocommunity/pack/rust/init.lua:60]: Interesting...I'll remove it! :D

Comment (Uzaaft) [reply lua/astrocommunity/pack/rust/init.lua:60]: Doesn't this mean that we can remove the whole opts function? @mrcjkb

Comment (mrcjkb) [reply lua/astrocommunity/pack/rust/init.lua:60]: I'm not sure what this part does:

```lua
return { server = require("astrolsp").lsp_opts "rust_analyzer", dap = { adapter = adapter } }
```

I suppose ` require("astrolsp").lsp_opts` includes an `on_attach` function that sets keymaps, etc.?

**Pull request merged**